[sha512] common: enable sha512 support

lsm5 opened the pull request:
<!--- Please read the [contributing guidelines](https://github.com/containers/container-libs/blob/main/CONTRIBUTING.md) before proceeding --->

Depends on (and includes changes from) #374 and #375  . Maybe better to let those go in separately before this one.

(cursor assisted).

mtrmac reviewed (commented):
Just an _extremely_ brief skim.

I strongly suspect that the existing approach of accepting _arbitraryRepo`:`_arbitrayTag_ and `sha256:`_digestValue_ in the same strings does not scale and is not sustainable . Doesn’t this break existing deployments where the repo name is `sha512`?

I don’t know what to actually _do_ here, mind.

mtrmac commented on common/libimage/pull.go:
```diff
@@ -261,7 +263,9 @@ func (r *Runtime) copyFromDefault(ctx context.Context, ref types.ImageReference,
-			imageName = "sha256:" + storageName[1:]
+			// Use the configured digest algorithm for the image name
+			digestAlgorithm := supportedDigests.Get()
+			imageName = digestAlgorithm.String() + ":" + storageName[1:]
```
(See below about `getImageID`)

lsm5 commented:
> I strongly suspect that the existing approach of accepting _arbitraryRepo_`:`_arbitraryTag_ and `sha256:`_digestValue_ in the same strings does not scale and is not sustainable . Doesn’t this break existing deployments where the repo name is `sha512`?

Added a `common/pkg/digestutils` package to handle this and other cases. With this vendored into buildah, and a Containerfile like:
```
FROM fedora@sha256:40f5b359d76ac10a563387e9e95efaa09db979b12b0ff8ad17bde1ac89c594b4
RUN dnf -y install bats
```
and `buildah build --digest sha512 -f $CONTAINERFILE -t sha512` , I get:
```
$ ./bin/buildah images --no-trunc 
REPOSITORY                          TAG      IMAGE ID                                                                                                                                  CREATED              SIZE
localhost/sha512                    latest   sha512:4f41c3cc5184119d2e6263ed6031948ae9f73613b245bfb13485bad562cd9e67ae42d7de9f6dd105fc811980418ca5d91976e60eff6048bf88e50637c94286d9   About a minute ago   306 MB
registry.fedoraproject.org/fedora   latest   sha256:6d1eb9616ac6246b3c2fe49ec5fef7a913eee5bf28e919f6dd8ea7604cc088c2                                                                   42 hours ago         170 MB
```

Let me know if that still doesn't address your concern.

Accounted for getImageID too and cleaned up the unnecessary tests.

PTAL. Thanks again!
